Questions about restrictions lint following update to clippy 0.1.67

### Summary

I am very new to Rust and thus, new to clippy and I've had troubles after updating clippy to version 0.1.67 (probably from version 0.1.65 or so) but this may show problems with my workflow more than problems with clippy itself.

1. My first question is about https://github.com/rust-lang/rust-clippy/pull/9755 .

Previously, in order to get the most out of clippy, I've tried to enable every single check I could. This involved using a super long list of arguments: "-D clippy::correctness  (...) -D clippy::restriction" and then a "whitelist" for restrictions check that I want to allow: "-A clippy::indexing_slicing -A clippy::integer_arithmetic -A clippy::arithmetic_side_effects (...)".
My idea was that if there were any checks added to clippy, they would be enabled by default and if they were to fail: I'd have to decide between adding the check to the list of lints allowed or fix things in my code (and learn something about good practices).

Now, it looks like using "-D clippy::restriction" is somehow forbidden and I do not quite understand the reasoning. Is using "-D clippy::restriction" with an explicit list of allowed lints frowned upon ? If so, should I instead maintained a list of disallowed lints and update it regularly based on the list of new checks in the documentation ? (I've tried to do this in a very tedious way before realising that I should have just used "-A clippy::blanket_clippy_restriction_lints".



2. My second question is about 2 fairly new lints: clippy::semicolon_inside_block and clippy::semicolon_outside_block

The documentation can be found on:

- https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_outside_block
- https://rust-lang.github.io/rust-clippy/master/index.html#semicolon_inside_block

As per the explanation above, I've tried to run every single lint including these 2 lints which are according to the documentation added in 1.66.0. However, they seem to be unknown on my version.

```
$ cargo clippy --version
clippy 0.1.67 (fc594f1 2023-01-24)
```

then

```
$ cargo clippy -- -D clippy::semicolon_outside_block
    Checking aoc v0.1.0 (/home/desodtsy/Documents/Geekage/aoc/rust)
error[E0602]: unknown lint: `clippy::semicolon_outside_block`
  |
  = note: requested on the command line with `-D clippy::semicolon_outside_block`

For more information about this error, try `rustc --explain E0602`.
```

then

```
$ cargo clippy -- -D clippy::semicolon_inside_block
    Checking aoc v0.1.0 (/home/desodtsy/Documents/Geekage/aoc/rust)
error[E0602]: unknown lint: `clippy::semicolon_inside_block`
  |
  = note: requested on the command line with `-D clippy::semicolon_inside_block`

For more information about this error, try `rustc --explain E0602`.
```



### Reproducer

_No response_

### Version

```text
$ rustc -Vv
rustc 1.67.0 (fc594f156 2023-01-24)
binary: rustc
commit-hash: fc594f15669680fa70d255faec3ca3fb507c3405
commit-date: 2023-01-24
host: x86_64-unknown-linux-gnu
release: 1.67.0
LLVM version: 15.0.6
```


### Additional Labels

_No response_

--- a/clippy_lints/src/semicolon_block.rs
+++ b/clippy_lints/src/semicolon_block.rs
@@ -30,7 +30,7 @@ declare_clippy_lint! {
     /// # let x = 0;
     /// unsafe { f(x); }
     /// ```
-    #[clippy::version = "1.66.0"]
+    #[clippy::version = "1.68.0"]
     pub SEMICOLON_INSIDE_BLOCK,
     restriction,
     "add a semicolon inside the block"
@@ -59,7 +59,7 @@ declare_clippy_lint! {
     /// # let x = 0;
     /// unsafe { f(x) };
     /// ```
-    #[clippy::version = "1.66.0"]
+    #[clippy::version = "1.68.0"]
     pub SEMICOLON_OUTSIDE_BLOCK,
     restriction,
     "add a semicolon outside the block"

